[Fix #187] Update docs to include phase_shift

chandrikarj opened the pull request:
## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [x] Other

**Why is this PR needed?**
This PR addresses the issue where the phase_shift step was missing from the documentation example. The neuropixels+kilosort2_5 configuration was updated to include the phase_shift step, but this was not reflected in the example, which was causing confusion.

**What does this PR do?**
This PR updates the documentation to include the missing phase_shift preprocessing step in the example configuration, making it consistent with the updated preprocessing pipeline.

## References
Issue #187 : Fix documentation example (this PR addresses the missing phase_shift step in the example)

## How has this PR been tested?
The updated documentation was built locally, and the example now includes the phase_shift step, ensuring consistency with the updated configuration. The documentation was verified by running the sphinx-build command.

## Is this a breaking change?
No

## Does this PR require an update to the documentation?
Yes, the documentation was updated to include the phase_shift preprocessing step in the example configuration. 

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

JoeZiminski reviewed (changes requested):
Thanks a lot for this @chandrikarj! This looks good, could you also add this to the `configs` example on both pages too? like:

```
configs = {
    "preprocessing": {
        "1": ["phase_shift", {}],
        "2": ["bandpass_filter", {"freq_min": 300, "freq_max": 6000}],
        "3": ["common_reference", {"operator": "median"}],
    }
}
```

This is a little confusing, but the functions can take the dict with the top level included or not, and so these examples just highlight that.

chandrikarj commented:
@JoeZiminski, I'm so sorry, I had changed the configs too but forgot to stage them, my bad. I have added it in a commit. Also the 01_preprocessing_sessions in the how-to section doesn't have configs right now, do you want me to add the configs column there like in tutorials?

JoeZiminski commented:
Hey @chandrikarj thanks a lot for this, no worries at all. For the how-to it's okay as it is (showing `pp_steps` dict and `"neuropixels+kilosort2_5"`), that is enough detail for a how-to.

This looks great, it is almost ready to merge, there is one error cropping up in build. To build the docs locally, you can first install the dependencies by `cd` to the repo and running:

```
pip install -r ./docs/requirements.txt
```

then `cd` into `docs` and run:

`make clean html`

the docs will be created in a folder called `html`, and you can open the main page by opening `index.html`.

When you run the command, you should see that an error is occurring, if you inspect the stack trace:

```
        Traceback (most recent call last):
          File "C:\Users\Jzimi\git-repos\forks\chandri\spikewrap\docs\source\galleries\tutorials\01_preprocessing_sessions.py", line 195, in <module>
            session.preprocess(
          File "C:\Users\Jzimi\git-repos\forks\chandri\spikewrap\spikewrap\structure\session.py", line 156, in preprocess
            run.preprocess(pp_steps, per_shank)
          File "C:\Users\Jzimi\git-repos\forks\chandri\spikewrap\spikewrap\structure\_run.py", line 125, in preprocess
            self._preprocessed[key] = Preprocessed(
                                      ^^^^^^^^^^^^^
          File "C:\Users\Jzimi\git-repos\forks\chandri\spikewrap\spikewrap\structure\_preprocessed.py", line 43, in __init__
            _fill_with_preprocessed_recordings(self._data, pp_steps)
          File "C:\Users\Jzimi\git-repos\forks\chandri\spikewrap\spikewrap\process\_preprocessing.py", line 42, in _fill_with_preprocessed_recordings
            preprocessed_recording = pp_funcs[pp_name](last_pp_step_output, **pp_options)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "C:\Users\Jzimi\miniconda3\envs\spikewrap8\Lib\site-packages\spikeinterface\preprocessing\filter.py", line 233, in __init__
            FilterRecording.__init__(
        TypeError: FilterRecording.__init__() got an unexpected keyword argument 'operator'
```
 you will see culprit line is:
```
          File "C:\Users\Jzimi\git-repos\forks\chandri\spikewrap\docs\source\galleries\tutorials\01_preprocessing_sessions.py", line 195, in <module>
```

This is because in the docs we do:
```
pp_attempt_2["preprocessing"]["2"][1]["operator"] = "average"
```

(which is not a very nice way to set these configs in general! and needs improving in future). The problem is because the numbers of the configs has changed, this should be set on step "3". So if you change the above line to:

pp_attempt_2["preprocessing"]["3"][1]["operator"] = "average"

The docs should build. I think this will then be ready to merge!

chandrikarj commented:
Fixed and tried it @JoeZiminski, the docs build now!🚀

JoeZiminski reviewed (approved):
Thanks @chandrikarj! this is good to go
